helper/resource/query/query_checks.go

```diff
@@ -28,6 +28,13 @@ func RunQueryChecks(ctx context.Context, t testing.T, query []tfjson.LogMsg, que
 
 	for _, msg := range query {
 		switch v := msg.(type) {
+		case tfjson.DiagnosticLogMessage:
+			if v.Diagnostic.Severity == tfjson.DiagnosticSeverityError {
+				// TODO reimplement missing .tf config error
+				result = append(result, fmt.Errorf("query error diagnostic - summary: %s, detail: %s", v.Diagnostic.Summary, v.Diagnostic.Detail))
+				continue
+			}
+
```

Comment:
Not a blocking comment, but food for thought. One way this differs from other types of checks (plan / state checks) is that if any of the other commands fail, they immediately return an error (usually from the `wd` method) which is then checked by `ExpectError` without running any checks.

Is an error diagnostic returned by a provider considered the end of a `query` command? Meaning that  running query checks on any results after would be considered invalid? Another way to achieve the same result here would just be returning the error diagnostic from `func (wd *WorkingDir) Query(ctx context.Context) ([]tfjson.LogMsg, error)`